src/optimization/optimalTau.go

```diff
@@ -0,0 +1,19 @@
+package optimization
+
+import "BGPAlert/blt_mad"
+
+func FindTauForMinReqOutput(dataArray []float64, minReqOutData []float64) float64 {
+	var tau float64 //need to find the minimum output array that contains all the results
+	tau = 0.0
+
+	for i := 0; i <= 1000; i++ { //check if I need to go above 100 but I am pretty sure that for most cases this
+		currentOutput := blt_mad.BltMad(dataArray, tau)
+		if blt_mad.ContainAllElements(currentOutput, minReqOutData) != nil { //if there are no missing elements; continue
```

Comment:
I think the result of this function is always not equal to `nil`. (it can be an empty slice though)
Also, if you're not using the actual elements returned by `ContainAllElements`, then maybe that function should only return true or false (and can be implemented more efficiently, returning as soon as it finds one element that is not in the main array)